tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/BulkheadAsynchTest.java

```diff
@@ -255,6 +267,41 @@ public void testBulkheadExceptionThrownWhenQueueFullAsync() throws InterruptedEx
         }
     }
 
+    /**
+     * Test that an asynchronous method which returns an incomplete CompletionStage still reserves a slot in the bulkhead
+     */
+    @Test
+    @SuppressWarnings("unused")
+    public void testBulkheadCompletionStage() throws InterruptedException, ExecutionException, TimeoutException {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        try {
+            Future<Void> future1 = toCompletableFuture(bhBeanCompletionStage.serviceCS(result));
+            Future<Void> future2 = toCompletableFuture(bhBeanCompletionStage.serviceCS(result));
+            Thread.sleep(getConfig().getTimeoutInMillis(200)); // Give tasks a chance to start and run
+            Future<Void> future3 = toCompletableFuture(bhBeanCompletionStage.serviceCS(result));
+            Future<Void> future4 = toCompletableFuture(bhBeanCompletionStage.serviceCS(result));
+            Thread.sleep(getConfig().getTimeoutInMillis(200)); // Give tasks a chance to start and run
+            Future<Void> future5 = toCompletableFuture(bhBeanCompletionStage.serviceCS(result));
+            
+            // Although futures 1 & 2 have had time to run, they shouldn't be finished because the completion stage returned is not complete
+            assertFalse(future1.isDone(), "Future1 reported done");
+            assertFalse(future2.isDone(), "Future2 reported done");
+            
+            // Because futures 1 & 2 are still not "complete" (even though the method call may have returned), future 5 should be rejected
+            expectBulkheadException(future5);
+            
+            // Complete the CompletionStage which was returned
+            result.complete(null);
+            
+            // After the CompletionStage completes, future 1 & 2 should complete
+            future1.get(getConfig().getTimeoutInMillis(1000), TimeUnit.MILLISECONDS);
+            future2.get(getConfig().getTimeoutInMillis(1000), TimeUnit.MILLISECONDS);
+        }
```

Comment:
Perhaps also check `future3` and `future4`? With Awaitility I guess :-)

Reply:
Done